packages/core/src/App/Containers/NotificationsDialog/notifications-dialog.jsx

```diff
@@ -5,147 +5,154 @@ import { CSSTransition } from 'react-transition-group';
+                                            'dc-btn',
+                                            'dc-btn--secondary',
+                                            'notifications-item__cta-button'
+                                        )}
+                                        to={item.action.route}
+                                    >
+                                        <Text weight='bold' size='xxs'>
+                                            {item.action.text}
+                                        </Text>
+                                    </BinaryLink>
+                                ) : (
+                                    <Button
+                                        className={classNames('dc-btn--secondary', 'notifications-item__cta-button')}
+                                        onClick={item.action.onClick}
+                                    >
+                                        <Text size='xxs'>{item.action.text}</Text>
+                                    </Button>
+                                )}
+                            </>
+                        )}
+                    </div>
+                </div>
+            ))
+        ) : (
+            <EmptyNotification />
+        )}
+    </React.Fragment>
+);
+
+const NotificationsDialogEl = React.forwardRef(({ notifications, toggleDialog }, ref) => {
+    const is_empty = !(notifications && notifications.length);
+    return (
+        <div className='notifications-dialog' ref={ref}>
+            <div className='notifications-dialog__header'>
+                <h2 className='notifications-dialog__header-text'>
+                    <Localize i18n_default_text='Notifications' />
+                </h2>
+            </div>
+            <div
+                className={classNames('notifications-dialog__content', {
+                    'notifications-dialog__content--empty': is_empty,
+                })}
+            >
+                <ThemedScrollbars is_bypassed={isMobile() || is_empty}>
+                    <NotificationsListEl notifications={notifications} toggleDialog={toggleDialog} />
+                </ThemedScrollbars>
+            </div>
+        </div>
+    );
+});
+NotificationsDialogEl.displayName = 'NotificationsDialogEl';
+
+const NotificationsDialog = ({ is_visible, notifications, toggleDialog }) => {
+    const wrapper_ref = React.useRef();
 
-    handleClickOutside = event => {
+    const handleClickOutside = event => {
         const notifications_toggle_btn = !event.target.classList.contains('notifications-toggle__icon-wrapper');
         if (
-            this.wrapper_ref &&
-            !this.wrapper_ref.contains(event.target) &&
-            this.props.is_visible &&
+            wrapper_ref.current &&
+            !wrapper_ref.current.contains(event.target) &&
+            is_visible &&
             notifications_toggle_btn
         ) {
-            this.props.toggleDialog();
+            toggleDialog();
         }
     };
```

Comment:
Is it possible replace `handleClickOutside` with the [hook](/packages/components/src/hooks/use-onclickoutside.js)?